### PR TITLE
Validate that extensions appear in correct messages

### PR DESCRIPTION
**Pull request by julek-wolfssl**

### CODE REVIEW

File: src/tls.c [hunk 1]
```diff
@@ -12452,13 +12512,9 @@ int TLSX_Parse(WOLFSSL* ssl, const byte* input, word16 length, byte msgType,
...
                 if (msgType != client_hello && msgType != session_ticket &&
-                        msgType != encrypted_extensions) {
-                    WOLFSSL_ERROR_VERBOSE(EXT_NOT_ALLOWED);
-                    return EXT_NOT_ALLOWED;
-                }
-                if (!IsAtLeastTLSv1_3(ssl->version) &&
-                        (msgType == session_ticket ||
-                         msgType == encrypted_extensions)) {
+                    msgType != encrypted_extensions &&
+                    msgType != session_ticket &&
```
Comment (SparkiDev) [inline src/tls.c:12516]: session_ticket and encrypted_extensions twice.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/tls.c (hunk 1 above)
Comment (julek-wolfssl) [reply src/tls.c]: Fixed.

File: src/tls.c [hunk 2]
```diff
@@ -12139,17 +12139,18 @@ int TLSX_Parse(WOLFSSL* ssl, const byte* input, word16 length, byte msgType,
...
-                    return EXT_NOT_ALLOWED;
+#ifdef WOLFSSL_TLS13
+                if (IsAtLeastTLSv1_3(ssl->version)) {
+                    if (msgType != client_hello &&
+                        msgType != encrypted_extensions)
+                        return EXT_NOT_ALLOWED;
                 }
+                else
 #endif
+                {
```
Comment (SparkiDev) [inline src/tls.c:12150]: Don't need braces.
And more below.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: src/tls.c (hunk 2 above)
Comment (julek-wolfssl) [reply src/tls.c:12150]: I think that the braces make future edits to the tls <= 1.2 path less error prone. Someone can add tls <= 1.2 specific logic without needing to remember about the tls 1.3 path.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**